nengo/params.py

```diff
@@ -292,6 +292,34 @@ def validate(self, instance, value):
         super(TupleParam, self).validate(instance, value)
 
 
+class ShapeParam(TupleParam):
+    """A parameter where the value is a tuple of integers."""
+
+    equatable = True
+
+    def __init__(self, name, default=Unconfigurable, length=None, low=0,
+                 optional=False, readonly=None):
+        super(ShapeParam, self).__init__(name, default=default, length=length,
+                                         optional=optional, readonly=readonly)
+        self.low = low
+
+    def __set__(self, instance, value):
+        try:
+            value = tuple(int(v) for v in value)
```

Comment:
My NumPy says about non-integer number in shapes "DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future". I would say, in anticipation of this, we should not allow non-integers at all.